src/api/Metadata.js

```diff
@@ -154,26 +176,223 @@ class Metadata extends File {
+        const filteredInstances = this.extractClassification(id, instances);
+
+        // Create editors from each instance
+        const editors: Array<MetadataEditor> = [];
+        await Promise.all(
+            filteredInstances.map(async instance => {
+                const template: ?MetadataEditorTemplate = await this.getTemplateForInstance(id, instance, templates);
+                if (template) {
+                    editors.push(this.createEditor(instance, template, canEdit));
+                }
             }),
-            'data.entries',
-            [],
         );
+        return editors;
+    }
+
+    /**
+     * API for getting metadata editors
+     *
+     * @param {string} fileId - Box file id
+     * @param {Function} successCallback - Success callback
+     * @param {Function} errorCallback - Error callback
+     * @param {boolean} hasMetadataFeature - metadata feature check
+     * @param {Object} options - fetch options
+     * @return {Promise}
+     */
+    async getMetadata(
+        file: BoxItem,
+        successCallback: ({ editors: Array<MetadataEditor>, templates: Array<MetadataEditorTemplate> }) => void,
+        errorCallback: ElementsErrorCallback,
+        hasMetadataFeature: boolean,
+        options: FetchOptions = {},
+    ): Promise<void> {
+        const { id, permissions, is_externally_owned }: BoxItem = file;
+        this.errorCode = ERROR_CODE_FETCH_METADATA;
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
+
+        // Check for valid file object.
+        // Need to eventually check for upload permission.
+        if (!id || !permissions) {
+            this.errorHandler(getBadItemError());
+            return;
+        }
+
+        const cache: APICache = this.getCache();
+        const key = this.getMetadataCacheKey(id);
+
+        // Clear the cache if needed
+        if (options.forceFetch) {
+            cache.unset(key);
+        }
+
+        // Return the cached value if it exists
+        if (cache.has(key)) {
+            this.successHandler(cache.get(key));
+            if (!options.refreshCache) {
+                return;
+            }
+        }
+
+        try {
+            const customPropertiesTemplate: MetadataEditorTemplate = this.getCustomPropertiesTemplate();
+            const [instances, globalTemplates, enterpriseTemplates] = await Promise.all([
+                this.getInstances(id),
+                this.getTemplates(id, METADATA_SCOPE_GLOBAL),
+                hasMetadataFeature ? this.getTemplates(id, METADATA_SCOPE_ENTERPRISE) : Promise.resolve([]),
+            ]);
+
+            const editors = await this.getEditors(
+                id,
+                instances,
+                customPropertiesTemplate,
+                enterpriseTemplates,
+                globalTemplates,
+                !!permissions.can_upload,
+            );
+
+            if (this.isDestroyed()) {
```

Comment:
Would something cause `isDestroyed()` to return true only now? Can this be checked earlier in the method?

Reply:
Needs to be after async. There are 2 asyncs going on, so it can additionally be slightly higher also, but has to remain here. I just didnt add it before since this one will take care of that too, though at the expense of a couple of wasted API request. Main goal is not to call successHandler, which this one achieves.

Reply:
I moved it just to guard successHandler based on another comment